web/views.py

```diff
@@ -38,33 +42,46 @@ def compare(request):
 
     lang1 = request.GET.get('lang1', '')
     lang2 = request.GET.get('lang2', '')
-    lang1_directory = meta_data_langs[lang1]
-    lang2_directory = meta_data_langs[lang2]
-
-    concept = request.GET.get('concept', '')
+    lang1_directory = meta_data_langs.get(lang1)
+    lang2_directory = meta_data_langs.get(lang2)
 
-    with open("web/thesauruses/" + lang1_directory + "/" + concept + ".json", 'r') as lang1_file:
-        data = lang1_file.read()
-    # parse file
-    lang1_file_json = json.loads(data)
-    lang1_concept = lang1_file_json[concept]
-    lang1_friendlyname = lang1_file_json["meta"]["language_name"]
+    if not (lang1_directory and lang2_directory):
+        return HttpResponseBadRequest("Lang does not exist")
 
-    with open("web/thesauruses/" + lang2_directory + "/" + concept + ".json", 'r') as lang2_file:
-        data = lang2_file.read()
-    # parse file
-    lang2_file_json = json.loads(data)
-    lang2_concept = lang2_file_json[concept]
-    lang2_friendlyname = lang2_file_json["meta"]["language_name"]
+    concept = request.GET.get('concept', '')
 
-    common_concepts =[]
-    for key in list(set.union(set(lang1_concept.keys()), set(lang2_concept.keys()))):
+    lang1_file_path = os.path.join(
+        "web", "thesauruses", lang1_directory, concept) + ".json"
+    lang2_file_path = os.path.join(
+        "web", "thesauruses", lang2_directory, concept) + ".json"
+
+    try:
+        with open(lang1_file_path, 'r') as lang1_file:
+            data = lang1_file.read()
+            # parse file
+            lang1_file_json = json.loads(data)
+            lang1_concept = lang1_file_json[concept]
+            lang1_friendlyname = lang1_file_json["meta"]["language_name"]
+
+        with open(lang2_file_path, 'r') as lang2_file:
+            data = lang2_file.read()
+            # parse file
+            lang2_file_json = json.loads(data)
+            lang2_concept = lang2_file_json[concept]
+            lang2_friendlyname = lang2_file_json["meta"]["language_name"]
+    except FileNotFoundError as fe:
+        return Http404
+
+    common_concepts = []
+    for key in (set(lang1_concept.keys()) & set(lang2_concept.keys())):
```

Comment:
I may need to test this. I need the equivalent of a union/full outer join on this and I can't tell if this will do the same thing. I need to match up language concepts if they both exist, but if one or the other doesn't exist, I still need it to show up and the other can be `null` or something. So I'll need to test this before approving.

Reply:
@geekygirlsarah I will save you some time This does not work that way, I thought you intended only to show the intersection i.e. the common concepts only. The union was throwing a key error, I will make the required changes here. I will make it show up as NA if it does not exist in one, hope that will be fine. Let me know about anything else that you would like to show instead.

Reply:
@geekygirlsarah I have updated it to show an unsupported message as follows: "Unsupported or Not Needed". Felt this was more apt.